src/utils/Store.tsx

```diff
@@ -0,0 +1,177 @@
+import { ErrorTypes } from '../types/ErrorTypes';
+
+interface Children {
+  children: React.ReactNode;
+}
+
+type ActiveUser = User | null;
+
+interface ActiveUserContextType {
+  activeUser: ActiveUser;
+  setActiveUser: React.Dispatch<SetStateAction<ActiveUser>> | undefined;
+}
+
+export const ActiveUserContext = createContext<ActiveUserContextType>({
+  activeUser: null,
+  setActiveUser: undefined,
+});
+
+export const ActiveUserProvider: React.FC<Children> = ({ children }) => {
+  const [activeUser, setActiveUser] = useState<ActiveUser>(null);
+
+  const value = useMemo(() => ({ activeUser, setActiveUser }), [activeUser]);
+
+  return (
+    <ActiveUserContext.Provider value={value}>
+      {children}
+    </ActiveUserContext.Provider>
+  );
+};
+
+interface Loader {
+  isLoading: boolean;
+  setIsLoading: React.Dispatch<SetStateAction<boolean>> | undefined;
+}
+
+export const LoaderContext = createContext<Loader>({
+  isLoading: false,
+  setIsLoading: undefined,
+});
+
+export const LoaderProvider: React.FC<Children> = ({ children }) => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const value = useMemo(
+    () => ({ isLoading, setIsLoading }),
+    [isLoading, setIsLoading],
+  );
+
+  return (
+    <LoaderContext.Provider value={value}>{children}</LoaderContext.Provider>
+  );
+};
+
+type PostsType = Post[] | null;
+
+interface Posts {
+  posts: PostsType;
+  setPosts: React.Dispatch<SetStateAction<PostsType>> | undefined;
+}
+
+export const PostsContext = createContext<Posts>({
+  posts: null,
+  setPosts: undefined,
+});
+
+export const PostsProvider: React.FC<Children> = ({ children }) => {
+  const [posts, setPosts] = useState<PostsType>(null);
+
+  const value = useMemo(() => ({ posts, setPosts }), [posts, setPosts]);
+
+  return (
+    <PostsContext.Provider value={value}>{children}</PostsContext.Provider>
+  );
+};
+
+type Error = ErrorTypes | null;
+
+interface Errors {
+  isError: Error;
+  setIsError: React.Dispatch<SetStateAction<Error>> | undefined;
+}
+
+export const ErrorsContext = createContext<Errors>({
+  isError: null,
+  setIsError: undefined,
+});
+
+export const ErrorsProvider: React.FC<Children> = ({ children }) => {
+  const [isError, setIsError] = useState<Error>(null);
+
+  const value = useMemo(() => ({ isError, setIsError }), [isError, setIsError]);
+
+  return (
+    <ErrorsContext.Provider value={value}>{children}</ErrorsContext.Provider>
+  );
+};
+
+type SideBarType = Post | null;
+
+interface IsActivePostType {
+  activePost: SideBarType;
+  setActivePost: React.Dispatch<SetStateAction<SideBarType>>;
+}
+
+export const ActivePostContext = createContext<IsActivePostType>({
+  activePost: null,
+  setActivePost: () => {},
+});
+
+export const ActivePostProvider: React.FC<Children> = ({ children }) => {
+  const [activePost, setActivePost] = useState<SideBarType>(null);
+
+  const value = useMemo(
+    () => ({ activePost, setActivePost }),
+    [activePost, setActivePost],
```

Comment:
You are using 'setActivePost' in the dependency array of useMemo, which is unnecessary because 'setActivePost' is a stable function and won't change between renders. Consider removing it from the dependency array.